alert-common/src/main/java/com/synopsys/integration/alert/common/message/model/ProviderMessageContent.java

```diff
@@ -33,18 +34,31 @@
 import com.synopsys.integration.builder.Buildable;
 
 public class ProviderMessageContent implements Buildable {
+    private static final String KEY_SEPARATOR = "_";
     private final LinkableItem provider;
     private final LinkableItem topic;
     private final LinkableItem subTopic;
+    private final String contentKey;
     private final Set<ComponentItem> componentItems;
 
-    private ProviderMessageContent(LinkableItem provider, LinkableItem topic, LinkableItem subTopic, Set<ComponentItem> componentItems) {
+    private ProviderMessageContent(LinkableItem provider, LinkableItem topic, LinkableItem subTopic, String contentKey, Set<ComponentItem> componentItems) {
         this.provider = provider;
         this.topic = topic;
         this.subTopic = subTopic;
+        this.contentKey = contentKey;
         this.componentItems = componentItems;
     }
 
+    public static final String generateContentKey(String providerName, String topicName, String topicValue, String subTopicName, String subTopicValue) {
```

Comment:
Do we want a ComponentKey object that has a string key and a "pretty print" method?

Reply:
That might be better then.  We have a MessageContentKey class we could follow the same pattern here to provide that pretty print.